allensdk/brain_observatory/ecephys/ecephys_session.py

```diff
@@ -1124,7 +1126,22 @@ def _build_stimulus_presentations(
         # pandas groupby ops ignore nans, so we need a new "nonapplicable"
         # value that pandas does not recognize as null ...
         stimulus_presentations.replace("", nonapplicable, inplace=True)
-        stimulus_presentations.fillna(nonapplicable, inplace=True)
+
+        # pandas does not automatically convert boolean cols for fillna
+        boolean_colnames = stimulus_presentations.dtypes[
+            stimulus_presentations.dtypes == "boolean"].index
+        col_type_map = {colname: "object" for colname in boolean_colnames}
+        stimulus_presentations = stimulus_presentations.astype(
+            col_type_map).fillna(nonapplicable)
```

Comment:
Okay, now I understand a bit better. `nonapplicable` is set to `"null"` hence the mixed type for the boolean columns need here. Remind me why we can't use the pandas NA value for these instead to preserve typing? Gonna guess something fails downstream?

Reply:
Line 1126 explains the decision to replace nan values to "null"
With pandas >1.4, the nan value of the boolean column is pd.NA by default.